Update Pontoon documentation resources.md

bcolsson opened the pull request:
Addresses #204

flodolo commented on src/tools/pontoon/resources.md:
```diff
@@ -4,30 +4,35 @@
-On the left side, above the entry, the name of the translator is visible in gray (linking to their profile), while in green it’s displayed how long ago the entry was submitted (full date and time is available as a tooltip). Below this information, the translation or suggestion is displayed.
+Each translation history entry contains:
+* The name of the translator and their profile picture (linking to their profile).
+* How long ago the entry was submitted (hover over to see the full date and time is available as a tooltip).
```
The sentence between parentheses is broken, the second part likely needs removing

flodolo commented on src/tools/pontoon/resources.md:
```diff
@@ -4,30 +4,35 @@
+* How long ago the entry was submitted (hover over to see the full date and time is available as a tooltip).
+* The translation or suggestion.
+* Icons indicating translation status (see below).
+* Translation comments. You can toggle `COMMENT` to show all comments for a translation or to add a new comment.
```
I think we can differentiate this. There will be the number of comments displayed, if there are any.

flodolo commented on src/tools/pontoon/resources.md:
```diff
@@ -4,30 +4,35 @@
 
-To the right of the entry, icons indicate the state of each element:
+Icons to the right indicate the status of each translation:
 * The green checkmark indicates that the translation has been approved.
```
Missing status for pretranslation.

flodolo commented on src/tools/pontoon/resources.md:
```diff
@@ -4,30 +4,35 @@
-If there are multiple suggestions for one string, it’s possible to use the `SHOW DIFF` link displayed on the right side of the suggestion: it will compare the selected suggestion to the first entry, highlighting in green all added text, in red all removed text. `HIDE DIFF` will display the original string.
+### Viewing Translation Differences
+
+The `DIFF` option appears if there are multiple suggestions for one string. Toggling `DIFF` compares the selected suggestion to the top entry, either the current approved translation or the most recent suggestion if no translation has been approved. Text hilighted in green indicates the text that has been added (compared to the top entry), text in red indicates removed text. Toggling `DIFF` again ...
```
Typo: hilighted

flodolo commented on src/tools/pontoon/resources.md:
```diff
@@ -4,30 +4,35 @@
 
 The Machinery tab shows possible translations from a variety of sources. These sources include:
-* Pontoon’s internal translation memory.
+* [Pontoon’s internal translation memory](../pontoon/translate.md#downloading-and-uploading-translations).
```
```suggestion
* [Pontoon’s internal translation memory](translate.md#downloading-and-uploading-translations).
```

Files are in the same folder.

flodolo commented:
One more note: the style that we used so far as been "impersonal" (passive vs you). If you search through the doc, all the "you" are from this edit.

I'm not sure if we have been 100% consistent across the documents, but this is an opportunity to fix that.

Thoughts?

flodolo commented:
I've started working on a couple of pages, and I'm trying to avoid having direct links to pontoon.mozilla.org where possible (since this documentation is for Pontoon as a whole). What I'm trying to do is use generic wording, and provide the Mozilla link as an example.

mathjazz commented:
Historically it made sense to cover History, Machinery and Locales in the same document, because they were located in the same part of the UI.

Since that's no longer the case, we should reorganize the doc accordingly.

Additionally, the name "Resource" is reserved for something else in Pontoon (project files).

flodolo commented:
The idea is to do a first pass to update the content, without drastically changing the structure.

Once content is updated, and most of the edits are out the way, I think it's going to be easier to reorganize content (e.g. moving this page elsewhere, maybe in ui.md)

flodolo commented:
In the end, I think this content will be integrated in `ui.md`. I'd leave out of this document COMMENTS and TERMS, as it is now, but I think we also need to explain what's "Concordance search".

bcolsson commented:
> One more note: the style that we used so far as been "impersonal" (passive vs you). If you search through the doc, all the "you" are from this edit.
> 
> I'm not sure if we have been 100% consistent across the documents, but this is an opportunity to fix that.
> 
> Thoughts?

Since it's already "impersonal" we should align with the existing style and avoid the usage of "you".

bcolsson commented:
> In the end, I think this content will be integrated in `ui.md`. I'd leave out of this document COMMENTS and TERMS, as it is now, but I think we also need to explain what's "Concordance search".

I've added a brief description + screen shot for Concordance Search under the Machinery tab section.

bcolsson commented:
Note: I'm going to do one more pass to see if any terms should be added once the glossary gets committed in #213.

mathjazz commented on src/assets/images/pontoon/resources/history.png:
It would be nice to reduce the amount of empty gray area in the source string panel.

mathjazz commented on src/tools/pontoon/resources.md:
```diff
@@ -2,35 +2,43 @@
-The history tab shows all of the suggestions and translations that have been submitted for the current source string.
-
-In the tab, the number of available entries is visible alongside the `HISTORY` title. The history entries are located below the tab.
+Pontoon displays the translation history for strings below the editor. The current approved translation for a source string appears at the top and all previous suggestions and translations beneath.
```
We also explain this (in slightly more detail) in line 29. Can we bring everything here?

bcolsson replied:
This part is laying out how the editor/translation history appear in Pontoon, while line 29 covers how the DIFF function works. I think the current order makes sense to introduce parts of translation history before going into detail on DIFF.

mathjazz commented on src/tools/pontoon/resources.md:
```diff
@@ -2,35 +2,43 @@
+Each translation history entry contains:
+* The name of the translator and their profile picture (linking to their profile).
+* How long ago the entry was submitted (hover over to see the full date and time as a tooltip).
+* The translation or suggestion.
```
Maybe just say "Translation"? https://github.com/mozilla-l10n/localizer-documentation/pull/213/files#diff-015370b13beb016806b36c39ecc5b6f21cfca5e9d63716f84c7aa563d5614bb2R29

mathjazz commented on src/tools/pontoon/resources.md:
```diff
@@ -2,35 +2,43 @@
 * [Microsoft Terminology](https://www.microsoft.com/Language/).
 * [Google Translate](https://translate.google.com).
 
+In addition, the user has the ability to search for translations containing words via [`Concordance Search`](#concordance-search)
```
Missing period.

mathjazz commented on src/tools/pontoon/resources.md:
```diff
@@ -2,35 +2,43 @@
 
+In addition, the user has the ability to search for translations containing words via [`Concordance Search`](#concordance-search)
+
 In the tab, the number of available entries is visible alongside the `MACHINERY` title.
```
Since the screenshot shows "2+1", we should clarify what that means.

mathjazz commented on src/tools/pontoon/resources.md:
```diff
@@ -41,9 +49,13 @@ Above the entry, a diff view between the current source string and the source st
 
 To the right of the entry, the number in green shows the percent match between the machinery string and the source string. The higher the percentage, the more likely the machinery is to be useful. The machinery source is listed in gray. Upon clicking the gray text, the machinery source will be opened in a new window. The green superscript to the right indicates the number of matches for the entry ...
```
The two paragraphs above no longer match the screenshots. The first one probably needs to be removed, because Machinery search has been replaced with Concordance search, which is covered below.

mathjazz commented on src/assets/images/pontoon/resources/locales.png:
We should update this screenshot to also contain preferred locales, because the green line is covered in the text.

bcolsson commented:
Current revision should address most of your concerns @flodolo / @mathjazz

flodolo commented on src/tools/pontoon/resources.md:
```diff
@@ -2,56 +2,68 @@
 * [Google Translate](https://translate.google.com).
 
-In the tab, the number of available entries is visible alongside the `MACHINERY` title.
+In addition, the user has the ability to search for translations containing words via [`Concordance Search`](#concordance-search).
```
```suggestion
In addition, the user has the ability to search for translations containing words via [`Concordance search`](#concordance-search).
```

You used lowercase later.

flodolo commented on src/tools/pontoon/resources.md:
```diff
@@ -2,56 +2,68 @@
+
+### Concordance search
+
+Concordance search allows users to search across all projects in Pontoon. Users can search for translations using strings in either source or target language. Matching results are displayed with the source string, translation, and project name and clicking a result will automatically fill the translation into the editing window. The search does not need to be related to the current project string ...
```
```suggestion
Concordance search allows users to search across all projects in Pontoon. Users can search for translations using strings in either source or target language. Matching results are displayed with the source string, translation, and project name; clicking a result will automatically fill the translation into the editor. Note that the search does not need to be related to the current string or project.
```

flodolo reviewed (changes requested):
Only a few nitpicks, nothing major.

We should optimize the PNGs, but I can do that via script before merging.

bcolsson commented:
Addressed all your comments. Added a slight tweak to the definition for locale.

> We should optimize the PNGs, but I can do that via script before merging.

I've been optimizing my screenshots using TinyPNG but feel free to optimize again.

mathjazz commented on src/tools/pontoon/resources.md:
```diff
@@ -2,62 +2,77 @@
+* [Pontoon’s internal translation memory](translate.md#downloading-and-uploading-translations).
 * [Microsoft Terminology](https://www.microsoft.com/Language/).
 * [Google Translate](https://translate.google.com).
+* [SYSTRAN](https://www.systran.net/)
+* [Caighdean](https://github.com/kscanne/caighdean)
+* [Bing Translator](https://www.bing.com/translator) (not currently enabled on pontoon.mozilla.org)
```
```suggestion
* [SYSTRAN](https://www.systran.net/).
* [Caighdean](https://github.com/kscanne/caighdean).
* [Bing Translator](https://www.bing.com/translator) (not currently enabled on pontoon.mozilla.org).
```

mathjazz commented on src/tools/pontoon/resources.md:
```diff
@@ -2,62 +2,77 @@
-### Stand-alone Machinery
+### Stand-alone machinery
 
 Pontoon machinery is also available as a [stand-alone web page](https://pontoon.mozilla.org/machinery/). To access the page, use the `Machinery` link available in the Profile menu, or the `Machinery` tab in the upper navigation displayed in some pages (e.g. *Teams page* or *Projects page*).
```
```suggestion
Pontoon machinery is also available as a [stand-alone web page](https://pontoon.mozilla.org/machinery/). To access the page, use the `Machinery` link in the main navigation (not accessible from the translation workspace).
```

mathjazz commented on src/tools/pontoon/resources.md:
```diff
@@ -2,62 +2,77 @@
 ![Locales tab](../../assets/images/pontoon/resources/locales.png)
 
-Next to the `LOCALES` title, the number of available entries is visible. The number of preferred locales is in green, all other locales are listed in gray.
+Next to the `LOCALES` title, the number of available entries is visible. The number of preferred locales are in green, all other locales are listed in gray. 
```
```suggestion
Next to the `LOCALES` title, the number of available entries is visible. The number of preferred locales are in green, all other locales are listed in gray.
```

mathjazz replied:
Curious - since it's just **one** "number of preferred locales", why **are**?
